sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -41,9 +41,18 @@ case class Project(projectList: Seq[NamedExpression], child: SparkPlan) extends
 
   @transient lazy val buildProjection = newMutableProjection(projectList, child.output)
 
-  protected override def doExecute(): RDD[InternalRow] = child.execute().mapPartitions { iter =>
-    val reusableProjection = buildProjection()
-    iter.map(reusableProjection)
+  override lazy val accumulators = Map(
+    "numRows" -> sparkContext.internalAccumulator(0L, "number of rows"))
+
+  protected override def doExecute(): RDD[InternalRow] = {
+    val numRows = accumulator[Long]("numRows")
```

Comment:
it is probably not useful actually to have numRows in project, since it is always one-to-one. it would be more useful to have numRows from data sources.